api/censuses.go

```diff
@@ -576,11 +638,16 @@ func (a *API) censusVerifyHandler(msg *apirest.APIdata, ctx *httprouter.HTTPCont
 	if err != nil {
 		return err
 	}
-	keyHash, err := ref.Tree().Hash(cdata.Key)
-	if err != nil {
-		return err
+
+	leafKey := cdata.Key
+	if ref.CensusType != int32(models.Census_ARBO_POSEIDON) {
+		leafKey, err = ref.Tree().Hash(cdata.Key)
+		if err != nil {
+			return err
+		}
 	}
-	valid, err := ref.Tree().VerifyProof(keyHash, cdata.Value, cdata.Proof, cdata.Root)
+
+	valid, err := ref.Tree().VerifyProof(leafKey, cdata.Value, cdata.Proof, cdata.Root)
 	if err != nil {
 		return err
```

Comment:
Please wrap the errors, not all of them but at least those that require explicit information such as `generated proof verification failed: %w`

Reply:
I included some messages but probably I need to review the code in deep to include more.